src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,198 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirstNode(value);
+        } else {
+            addNodeToEnd(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkElementIndexForAdd(index);
+        if (size == 0) {
+            addFirstNode(value);
+        } else if (index == 0) {
+            addNodeToStart(value);
+        } else if (index == size) {
+            addNodeToEnd(value);
+        } else {
+            Node<T> currentNode = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        return node.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> nodeByIndex = findNodeByIndex(index);
+        T newValue = nodeByIndex.element;
+        nodeByIndex.element = value;
+        return newValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (x.element == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (object.equals(x.element)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addNodeToStart(T value) {
+
+        Node<T> newNode = new Node<>(first, value, null);
+        first.prev = newNode;
+        first = newNode;
+        size++;
+    }
+
+    private void addNodeToEnd(T value) {
+
+        Node<T> newNode = new Node<>(last, value, null);
+        last.next = newNode;
+        last = newNode;
+        size++;
+    }
+
+    private void checkElementIndex(int index) {
```

Comment:
Create informative variable and method names. The name 'checkElementIndex' does not clearly indicate that this method is for checking the index for 'get', 'set', and 'remove' operations. Consider renaming to something like 'checkIndexExists' or 'validateIndexInRange'.